src/SignUpPage.tsx

```diff
@@ -0,0 +1,200 @@
+`;
+
+function validEmail(email: string) {
+	return !!email.match(/.+@.+\..{2,}/);
+}
+
+function compareEmail(email1: string, email2: string) {
+	return !!email1.match(email2);
+}
+
+// to do
+function validPass(password: string) {
+	return !!password.match(/.+@.+\..{2,}/);
+}
+
+function comparePass(password1: string, password2: string) {
+	return !!password1.match(password2);
+}
+
+function SignUpPage() {
+	const [email, setEmail] = useState('');
+	const [confirmEmail, setConfirmEmail] = useState('');
+	const [username, setUsername] = useState('');
+	const [emailErr, setEmailErr] = useState(false);
+	const [userPass, setUserPass] = useState('');
+	const [confirmUserPass, setConfirmUserPass] = useState('');
+	const [passErr, setPassErr] = useState(false);
+
+	const onSubmit = () => {
+		console.log('Creating Account.');
+		console.log(`Username: ${email}`);
+		console.log(`Password: ${userPass.replaceAll(/.{1}/g, '*')}`);
+		if (!validEmail(email)) {
+			setEmailErr(true);
+			return;
+		}
+	};
+
+	return (
+		<LoginPageContainer>
+			<LoginContents>
+				<PassrLogo>
+					<img src={logo} alt="logo" />
+				</PassrLogo>
+
+				<InputContainer>
+					<InputLabel>
+						<label>Email *</label>
+					</InputLabel>
+					<InputBox
+						type="text"
+						value={email}
+						onChange={(e) => {
+							if (validEmail(e.target.value)) setEmailErr(false);
+							setEmail(e.target.value);
+						}}
+						onBlur={(e) => {
+							if (!validEmail(e.target.value)) setEmailErr(true);
+						}}
+					/>
+				</InputContainer>
+
+				<InputContainer>
+					<InputLabel>
+						<label>Confirm Email *</label>
+					</InputLabel>
+					<InputBox
+						type="text"
+						value={confirmEmail}
+						onChange={(e) => {
+							if (compareEmail(email, confirmEmail))
+								setEmailErr(false);
+							setConfirmEmail(e.target.value);
+						}}
+						onBlur={(e) => {
+							if (!compareEmail(email, confirmEmail))
+								setEmailErr(true);
+						}}
+					/>
+				</InputContainer>
+
+				<InputContainer>
+					<InputLabel>
+						<label>Username *</label>
+					</InputLabel>
+					<InputBox
+						type="text"
+						value={confirmEmail}
+						onChange={(e) => {
+							setUsername(e.target.value);
+						}}
+					/>
+				</InputContainer>
+
+				<InputContainer>
+					<InputLabel>
+						<label className="Password-label">Password *</label>
+					</InputLabel>
+					<InputBox
+						type="password"
+						value={userPass}
+						onChange={(e) => {
+							if (validPass(e.target.value)) setPassErr(false);
+							setUserPass(e.target.value);
+						}}
+					/>
+				</InputContainer>
+
+				<InputContainer>
+					<InputLabel>
+						<label className="Password-label">
+							Confirm Password *
+						</label>
+					</InputLabel>
+					<InputBox
+						type="password"
+						value={confirmUserPass}
+						onChange={(e) => {
+							if (comparePass(userPass, confirmUserPass))
+								setPassErr(false);
+							setConfirmUserPass(e.target.value);
+						}}
```

Comment:
Can we add an `Onblur` event here to set `passErr` to true if the passwords don't match?  Then add `disabled = {emailErr || passErr} to the `LoginButton` below?